### ISSUE TITLE
/api/v2.0/domains/name/:name no longer accepts a literal "." in domain names, probably other routes too

### ISSUE THREAD

**Issue by makyen** [Charcoal-SE/metasmoke#880]
Issue reported in Charcoal HQ [here](https://chat.stackexchange.com/transcript/11540?m=59161580#59161580) by tripleee, and response [here](https://chat.stackexchange.com/transcript/11540?m=59162008#59162008).

The `/domains/name/:name` route and at least a one other route, `/posts/uid`, needs to accept `.` characters in the variable portion of the route. This was previously a problem, for which we implemented a workaround (see PR #851, and a [similar change to `/posts/uid`](https://github.com/Charcoal-SE/metasmoke/commit/025e12845e321b226c3ba07db92604a7bc907626) in [posts_api.rb](https://github.com/Charcoal-SE/metasmoke/blob/master/app/api/api/posts_api.rb#L86)), but it is now non-functional.

**Comment by user12986714** [comment]
From [Grape readme](https://github.com/ruby-grape/grape#anchoring):
> There is one caveat though: the params[:status] parameter only holds the first part of the request url. Luckily this can be circumvented by using the described above syntax for path specification and using the PATH_INFO Rack environment variable, using env['PATH_INFO']. This will hold everything that comes after the '/statuses/' part.

tl;dr: Unlike Rails routing, we probably need to parse `env['PATH_INFO']` ourselves. Git blame suggests that these lines in readme.md has not been touched since 6 years ago, so I have no idea why the original workaround even worked to begin with.
